lib/aws-record/record/item_operations.rb

```diff
@@ -95,26 +112,58 @@ def assign_attributes(opts)
       end
 
       # Mass assigns the attributes to the model and then performs a save
+      #
+      # You can use the +:force+ option to perform a simple put/overwrite
+      # without conditional validation or update logic.
+      #
+      #
+      # @example Usage Example
+      #   class MyModel
+      #     include Aws::Record
+      #     integer_attr :id,   hash_key: true
+      #     string_attr  :name, range_key: true
+      #   end
+      #
+      #   model = MyModel.new(id: 4, name: "John")
+      #   model.name  # => "John"
+      #   model.save   
+      # 
+      #   model.update(name: "Janeß")
+      #   model.name       # => "Jane"
+      #   model.persisted? # => true
       # 
+      # @param [Hash] new_param, contains the new parameters for the model
+      #
       # @param [Hash] opts
-      def update(opts)
-        self.assign_attributes(opts)
-
-        self.save
+      # @option opts [Boolean] :force if true, will save as a put operation and
+      #  overwrite any existing item on the remote end. Otherwise, and by
+      #  default, will either perform a conditional put or an update call.
+      # @return false if the record is invalid as defined by an attempt to call
+      #  +valid?+ on this item, if that method exists. Otherwise, returns client
+      #  call return value.
+      def update(new_params, opts = {})
+        assign_attributes(new_params)
+        save(opts)
       end
 
       # Updates model attributes and validates new values
       #
+      # You can use the +:force+ option to perform a simple put/overwrite
```

Comment:
This actually raises some interesting semantic questions. In ActiveRecord, they generally step around this because the ID field is a semi-hidden field. Where here, `#update` could be a `put_item` operation depending on the context. I don't think this is inherently wrong, but we either want to:

1. Call this out more explicitly in documentation.
2. Prevent the changing of key attributes in `#assign_attributes`, ensuring a pure update style.

Thoughts?

Reply:
While it initially seems wrong for the keys to be mutable, the current implementation is consistent with what happens if the user manually changes the key value and performs a save.

Reply:
I agree with you. I just think we should be explicit about this, because it has a chance to be a surprise to some users.